pkg/collector/python/datadog_agent.go

```diff
@@ -609,3 +612,78 @@ func ObfuscateMongoDBString(cmd *C.char, errResult **C.char) *C.char {
 	// memory will be freed by caller
 	return TrackedCString(obfuscatedMongoDBString)
 }
+
+var (
+	telemetryHistograms = new(sync.Map)
+)
+
+func lazyInitTelemetryHistogram(checkName string, metricName string) telemetry.Histogram {
+	var key = checkName + "." + metricName
+	histogram, _ := telemetryHistograms.LoadOrStore(key, telemetryimpl.GetCompatComponent().NewHistogramWithOpts(
+		checkName,
+		metricName,
+		nil,
+		fmt.Sprintf("Histogram of %s for Python check %s", metricName, checkName),
+		[]float64{10, 25, 50, 75, 100, 250, 500, 1000, 10000},
+		telemetry.DefaultOptions,
+	))
+	return histogram.(telemetry.Histogram)
+}
+
+var (
+	telemetryCounters = new(sync.Map)
+)
+
+func lazyInitTelemetryCounter(checkName string, metricName string) telemetry.Counter {
+	var key = checkName + "." + metricName
+	counter, _ := telemetryCounters.LoadOrStore(key, telemetryimpl.GetCompatComponent().NewCounterWithOpts(
+		checkName,
+		metricName,
+		nil,
+		fmt.Sprintf("Counter of %s for Python check %s", metricName, checkName),
+		telemetry.DefaultOptions,
+	))
+	return counter.(telemetry.Counter)
+}
+
+var (
+	telemetryGauges = new(sync.Map)
+)
+
+func lazyInitTelemetryGauge(checkName string, metricName string) telemetry.Gauge {
+	var key = checkName + "." + metricName
+	gauge, _ := telemetryGauges.LoadOrStore(key, telemetryimpl.GetCompatComponent().NewGaugeWithOpts(
+		checkName,
+		metricName,
+		nil,
+		fmt.Sprintf("Gauge of %s for Python check %s", metricName, checkName),
+		telemetry.DefaultOptions,
+	))
+	return gauge.(telemetry.Gauge)
+}
+
+// EmitAgentTelemetry records a telemetry data point for a Python integration
+// NB: Cross-org agent telemetry needs to be enabled for each metric in
+// comp/core/agenttelemetry/impl/config.go defaultProfiles
+//
+//export EmitAgentTelemetry
+func EmitAgentTelemetry(checkName *C.char, metricName *C.char, metricValue C.float, metricType *C.char) {
+	goCheckName := C.GoString(checkName)
+	goMetricName := C.GoString(metricName)
+	goMetricValue := float64(metricValue)
+	goMetricType := C.GoString(metricType)
+
+	switch goMetricType {
+	case "counter":
+		counter := lazyInitTelemetryCounter(goCheckName, goMetricName)
+		counter.Add(goMetricValue)
+	case "histogram":
+		histogram := lazyInitTelemetryHistogram(goCheckName, goMetricName)
+		histogram.Observe(goMetricValue)
+	case "gauge":
+		gauge := lazyInitTelemetryGauge(goCheckName, goMetricName)
+		gauge.Add(goMetricValue)
+	default:
+		log.Warnf("EmitAgentTelemetry: unsupported metric type %s", goMetricType)
+	}
+}
```

Comment:
Agree it should be an enum but AFAIK there's no way to enforce that in Python. In practice since it's a small number of strings that are familiar to people I think it'll be ok.

I think we can leave tagging for a follow-up given its complexity and that we don't necessary want a lot of tags on these metrics.